frontend/server/src/Controllers/Certificate.php

```diff
@@ -49,36 +49,28 @@ public static function apiGenerateContestCertificates(\OmegaUp\Request $r) {
         }
 
         //check if is a certificate generator
-        if (\OmegaUp\Authorization::isCertificateGenerator($r->identity)) {
-            if ($contest->certificates_status === 'uninitiated' || $contest->certificates_status === 'retryable_error') {
-                // add certificates_cutoff value to the course
-                if (!is_null($r['certificates_cutoff'])) {
-                    $contest->certificate_cutoff = $r['certificates_cutoff'];
-                }
-                // update contest with the new value
-                \OmegaUp\DAO\Contests::update($contest);
-
-                //connection to rabbitmq
-                $connection = new \PhpAmqpLib\Connection\AMQPStreamConnection(
-                    'rabbitmq',
-                    5672,
-                    'omegaup',
-                    'omegaup'
-                );
+        if (!\OmegaUp\Authorization::isCertificateGenerator($r->identity)) {
+            throw new \OmegaUp\Exceptions\ForbiddenAccessException();
+        }
+        if ($contest->certificates_status === 'uninitiated' || $contest->certificates_status === 'retryable_error') {
```

Comment:
acá también conviene usar el guard clause pattern:

```suggestion
        if ($contest->certificates_status !== 'uninitiated' && $contest->certificates_status !== 'retryable_error') {
            return ['status' => 'ok'];
        }
```